src/ophyd_async/fastcs/eiger/_eiger.py

```diff
@@ -26,7 +26,7 @@ def __init__(
         name="",
     ):
         self.drv = EigerDriverIO(prefix + drv_suffix)
-        self.odin = Odin(prefix + hdf_suffix + "FP:")
+        self.odin = OdinHdfIO(prefix + hdf_suffix + "FP:")
```

Comment:
As discussed in person, I wonder if this should be structured to point to the top level controller of the FastCS-odin. The we would have `det.odin.fp.hdf.file_path` rather than `det.odin.file_path`, in case there are other things to go in `fr` or other plugins in `fp` we need to reference later. OdinWriter can still take an `OdinHdfIO` in its constructor if all the PVs it needs are in that Device